packages/block-library/src/search/view.js

```diff
@@ -66,6 +81,39 @@ const { actions } = store(
 					actions.closeSearchInput();
 				}
 			},
+			*updateSearch() {
+				const { ref } = getElement();
+				const { value } = ref;
+
+				// Don't navigate if the search didn't really change.
+				if ( value === state.search ) {
+					return;
+				}
+
+				state.search = value;
+
+				// Debounce the search by 300ms to prevent multiple navigations.
+				// We can do this by yielding a promise that resolves after 300ms and
+				// then bailing out if the search has changed.
+				yield new Promise( ( resolve ) => setTimeout( resolve, 300 ) );
+				if ( value !== state.search ) {
+					return;
+				}
+
+				const url = new URL( window.location.href );
+				if ( ! isEmpty( value ) ) {
+					state.search = value;
+					url.searchParams.set( 'instant-search', value );
+				} else {
+					url.searchParams.delete( 'instant-search' );
+				}
```

Comment:
I'm pretty sure we want to clear `paged` param whenever navigate is called. It doesn't make sense to replace the search results and start at another page that may or may not even exist.

Instead of `isEmpty`, a falsy check is done on the `value` string.

```suggestion
				const url = new URL( window.location.href );
				url.searchParams.delete( 'paged' );
				if ( value ) {
					url.searchParams.set( 'instant-search', value );
				} else {
					url.searchParams.delete( 'instant-search' );
				}
```